### PR TITLE
feat: Update Leanplum SDK to Version 6.x

### PR DESCRIPTION
**Pull request by BrandonStalnaker**
## Summary
 - Updated to latest version of Leanplum

 ## Testing Plan
 - Tested locally and confirmed no breaking changes

 ## Reference Issue (For mParticle employees only.  Ignore if you are an outside contributor)
 - Closes https://go.mparticle.com/work/SQDSDKS-5849

### CODE REVIEW

File: mParticle-Leanplum.podspec [hunk 1]
```diff
@@ -16,7 +16,7 @@ Pod::Spec.new do |s|
     s.ios.deployment_target = "9.0"
     s.ios.source_files      = 'mParticle-Leanplum/*.{h,m,mm}'
     s.ios.dependency 'mParticle-Apple-SDK/mParticle', '~> 8.0'
-    s.ios.dependency 'Leanplum-iOS-SDK', '~> 4.1'
+    s.ios.dependency 'Leanplum-iOS-SDK', '~> 6.0'
```
Comment (jasonmgeorge) [inline mParticle-Leanplum.podspec]: Any reason not to align on 6.1 here?

Comment (BrandonStalnaker) [reply mParticle-Leanplum.podspec]: @jasonmgeorge It'll automatically take anything up to 6.x with this. I could put 6.1 to make that the minimum but I think keeping it .0 helps people remember the "up to next major version" bit better

Comment (via-guy) [reply mParticle-Leanplum.podspec]: It would make sense to be aligned with the others, though. No?

Comment (einsteinx2) [reply mParticle-Leanplum.podspec]: @via-guy agreed, we should either change this one to 6.1 or change the others to 6.0 and 6.0.0 respectively

**Review by einsteinx2 — CHANGES_REQUESTED** [review]

**Review by einsteinx2 — APPROVED** [review]
lgtm

**Pull request merged**